py_gtfs_rt_ingestion/tests/test_batcher.py

```diff
@@ -0,0 +1,99 @@
+        assert batch.create_event() == {'files':[]}
+
+    # `add_file` method operating correctly 
+    files = {
+        'test100': 100,
+        'test200': 200,
+    }
+    config_type = ConfigType.RT_VEHICLE_POSITIONS
+    batch = Batch(config_type=config_type)
+    for filename, filesize in files.items():
+        batch.add_file(filename=filename, filesize=filesize)
+    # Checking Batch __str__ method
+    print(batch)
+    out, err = capfd.readouterr()
+    assert out == f"Batch of {sum(files.values())} bytes in {len(files)} {config_type} files\n" 
+
+    # Calling `trigger_lambda` method raises exception
+    with pytest.raises(NoImplException):
+        Batch(ConfigType.RT_VEHICLE_POSITIONS).trigger_lambda()
+
+def test_bad_file_names():
+    # Check `batch_files` handling of bad filenames
+    files = [('test1',0),('test2',1),(None,0),(None,1)]
+    assert [b for b in batch_files(files=files, threshold=100)] == []
+
+    # Check mix of good and bad filenames:
+    files = [
+                ('test1',0),
+                (None,100),
+                ('https_cdn.mbta.com_realtime_VehiclePositions_enhanced.json.gz',100)
+            ]
+    batches = [b for b in batch_files(files=files, threshold=100)]
+    assert len(batches) == 1
+
+def test_empty_batch():
+    # Check `batch_files` handling of empty iterator
+    assert [b for b in batch_files(files=[], threshold=100)] == []
+
+def test_batch_files(s3_stub):
+    threshold = 100_000
+
+    files = [
+        ('https_cdn.mbta.com_realtime_VehiclePositions_enhanced.json.gz', 100_000),
+    ]
+    batches = [b for b in batch_files(files=files, threshold=threshold)]
+    assert len(batches) == len(files)
+
+    files = [
+        ('https_cdn.mbta.com_realtime_VehiclePositions_enhanced.json.gz', threshold),
+        ('https_cdn.mbta.com_realtime_VehiclePositions_enhanced.json.gz', threshold),
+    ]
+    batches = [b for b in batch_files(files=files, threshold=threshold)]
+    assert len(batches) == len(files)
+    for b in batches:
+        assert b.total_size == threshold
+
+    # Process large page_obj_response from json file 'test_files/large_page_obj_response.json'
+    # large json file contains 1,000 Contents records.
+    large_response_file = os.path.join(TEST_FILE_DIR,"large_page_obj_response.json")
+    page_obj_params = {
+        "Bucket": 'mbta-gtfs-s3',
+        "Prefix": ANY,
+    }
+    with open(large_response_file, 'r') as f:
+        page_obj_response = json.load(f)
+    s3_stub.add_response("list_objects_v2", page_obj_response, page_obj_params)
+    with s3_stub:
+        files = [file for file in file_list_from_s3('mbta-gtfs-s3','')]
+
+    batches = [b for b in batch_files(files=files, threshold=threshold)]
+    # Verify expected number of batches produced
+    assert len(batches) == 20
+    # Verify each batch total_size respects threshold
+    for b in batches:
+        assert b.total_size <= threshold
```

Comment:
this should probably be wrapped in an `if len(b.files) > 1` gate, since we decided to allow batches to contain a single file if they exceed threshold size.